pages/home_page.py

```diff
@@ -24,6 +26,13 @@ def __init__(self, testsetup, product=None):
 
         if product is None:
             self.selenium.get(self.base_url)
+        self.wait_for_page_to_load()
+
+    def wait_for_page_to_load(self):
+        Wait(self.selenium, self.timeout).until(
+            lambda s: 'loading' not in s.find_element(
+                *self._graph_locator).get_attribute('class'))
+        return self
```

Comment:
There's a bit of inconsistency between our suites on this. Recently I have been explicitly implementing this wait method and an open method so that instantiation is not tied with any interactions. This allows us to wait for the page to load even if we're creating it as a result of calling a method in another page object, instead of opening the page directly.

I intend to standardise this in the future by offering a page object model package that can be used and extended by our suites.

> On 23 Nov 2015, at 19:33, Matt Brandt <notifications@github.com> wrote:
> 
> In pages/home_page.py:
> 
> > @@ -24,6 +26,13 @@ def __init__(self, testsetup, product=None):
> >  
> >          if product is None:
> >              self.selenium.get(self.base_url)
> > +        self.wait_for_page_to_load()
> > +
> > +    def wait_for_page_to_load(self):
> > +        Wait(self.selenium, self.timeout).until(
> > +            lambda s: 'loading' not in s.find_element(
> > +                *self._graph_locator).get_attribute('class'))
> > +        return self
> I like this approach. One additional question, have we standardized on moving the wait_for_page_to_load() out of the constructor?
> 
> —
> Reply to this email directly or view it on GitHub.
>